app/controller/command/commands/karma.py

```diff
@@ -0,0 +1,154 @@
+        parser_view.set_defaults(which="view",
+                                 help="view a user's karma amount")
+        parser_view.add_argument("slack_id", metavar="SLACK-ID",
+                                 type=str, action='store',
+                                 help="slack id of user karma to view")
+        return subparsers
+
+    def handle(self, command, user_id):
+        """Handle command by splitting into substrings."""
+        logging.info('Handling karma Command')
+        command_arg = shlex.split(command)
+        args = None
+
+        try:
+            args = self.parser.parse_args(command_arg)
+        except SystemExit:
+            return self.get_help(), 200
+
+        if args.which == "set":
+            return self.set_helper(user_id, args.slack_id, args.amount)
+        elif args.which == "reset":
+            return self.reset_helper(user_id, args.all)
+        elif args.which == "view":
+            return self.view_helper(user_id, args.slack_id)
+        else:
+            return self.get_help(), 200
+
+    def add_karma(self, giver_id, receiver_id) -> ResponseTuple:
+        """Give karma from giver_id to receiver_id."""
+        logging.info("giving karma to " + receiver_id)
+        if(giver_id == receiver_id):
+            return "cannot give karma to self", 200
+        try:
+            user = self.facade.retrieve(User, receiver_id)
+            user.karma += self.karma_add_amount
+            self.facade.store(user)
+            return f"gave 1 karma to {user.name}", 200
+        except LookupError:
+            return self.lookup_error, 200
+
+    def get_help(self) -> str:
+        """Return command options for team events."""
+        res = "\n*" + self.command_name + " commands:*```"
+        for argument in self.subparser.choices:
+            name = argument.capitalize()
+            res += "\n*" + name + "*\n"
+            res += self.subparser.choices[argument].format_help()
+        return res + "```"
+
+    def set_helper(self,
+                   user_id: str,
+                   slack_id: str,
+                   amount: int) -> ResponseTuple:
+        """Manually sets a user's karma."""
+        try:
+            user_command = self.facade.retrieve(User, user_id)
+            if user_command.permissions_level == Permissions.admin:
+                user = self.facade.retrieve(User, slack_id)
+                user.karma = amount
+                self.facade.store(user)
+                return f"set {user.name}'s karma to {amount}", 200
+            else:
+                return self.permission_error, 200
+        except LookupError:
+            return self.lookup_error, 200
+
+    def reset_helper(self,
+                     user_id: str,
+                     reset_all: bool) -> ResponseTuple:
+        """Reset all user's karma."""
+        try:
+            user_command = self.facade.retrieve(User, user_id)
+            if user_command.permissions_level == Permissions.admin:
```

Comment:
I don't like too many indents. Invert these if statements so that you return immediately, and proceed as normal afterwards.

Turn:

```python
def foo(a: bool, b: bool) -> str:
    if a:
        if b:
            return "BEE"
        else:
            return "CEE"
    return "ERROR"
```

Into:

```python
def foo(a: bool, b: bool) -> str:
    if not a:
        return "ERROR"

    if b:
        return "BEE"
    else:
        return "CEE"

```